experimental/Matrix/matrix.g

```diff
@@ -0,0 +1,192 @@
+                Print("  ");
+            else
+                Print(" ] ] \n");
+            fi;
+        od;
+    end);
+
+InstallOtherMethod(Unpack, [IsJuliaMatrixRep], function(m)
+    local v, i, j;
+    v := EmptyPlist(NumberRows(m)*NumberColumns(m));
+    for i in [1..NumberRows(m)] do
+        for j in [1..NumberColumns(m)] do
+            v[j+NumberColumns(m)*(i-1)] := m[i,j];
+        od;
+    od;
+    return v;
+end);
+
+
+# Method: SetNiceMorphismForJuliaMatrixRepGroup
+# Input: Matrix group G
+# Output: -
+# This function sets a nice morphism for G. The morphism is defined from the Julia matrix to the corresponding GAP matrix
+
+DeclareOperation("SetNiceMorphismForJuliaMatrixRepGroup", [IsGroup]);
+
+InstallMethod(SetNiceMorphismForJuliaMatrixRepGroup, [IsGroup], function(G)
+    local hom, f, f_inv, ele, GAPGenerators, i, gens, GAPGroup, JuliaGAPMap;
+    
+        gens := GeneratorsOfGroup(G);
+        Assert(0, ForAll(gens, IsJuliaMatrixRep));
+        
+        ele := gens[1];
+        hom := Julia.Oscar._iso_oscar_gap(ele!.m.base_ring);
+        
+        GAPGenerators := List(gens, g -> Julia.AbstractAlgebra.map_entries(hom, g!.m));
+        
+        GAPGroup := GroupByGenerators(GAPGenerators);
+        
+        f := function(m) return Julia.AbstractAlgebra.map_entries(hom,m!.m); end;
+        f_inv := function(m) return MakeJuliaMatrixRep(Julia.Oscar.preimage_matrix(hom,m)); end;
+
+        JuliaGAPMap := GroupHomomorphismByFunction(G,GAPGroup,f,f_inv);
+        SetNiceMonomorphism(G,JuliaGAPMap);
+        SetIsHandledByNiceMonomorphism(G, true);
+    end);
+    
+InstallOtherMethod(\=, [IsJuliaMatrixRep, IsJuliaMatrixRep], function( m1, m2 )
+        return m1!.m = m2!.m;
+    end);
+    
+# TODO: Not working in all cases. Needs more checks?
+InstallOtherMethod(\<, [IsJuliaMatrixRep, IsJuliaMatrixRep], function( m1, m2 )
+        return m1!.m < m2!.m;
+    end);
+
+InstallOtherMethod(MinimalPolynomial, [IsJuliaMatrixRep], function(m)
+    return Julia.AbstractAlgebra.minpoly(m!.m);
+end);
+
+InstallOtherMethod(CharacteristicPolynomial, [IsJuliaMatrixRep], function(m)
+    return Julia.AbstractAlgebra.charpoly(m!.m);
+end);
+
+InstallOtherMethod(IsSymmetric, [IsJuliaMatrixRep], function(m)
+    return Julia.AbstractAlgebra.issymmetric(m!.m);
+end);
+
+InstallOtherMethod(IsMonomial, [IsJuliaMatrixRep], function(m)
+    return Julia.AbstractAlgebra.ismonomial(m!.m);
+end);
+
+InstallOtherMethod(Rank, [IsJuliaMatrixRep], function(m)
+    return Julia.AbstractAlgebra.rank(m!.m);
+end);
+
+InstallOtherMethod(Trace, [IsJuliaMatrixRep], function(m)
+    return Julia.AbstractAlgebra.tr(m!.m);
+end);
+
+
+DeclareOperation("TransformPolynomialFromJuliaToGAP", [IsJuliaObject]);
+
+InstallMethod(TransformPolynomialFromJuliaToGAP, [IsJuliaObject], function(pol)
+    local x, hom, res, i;
+    
+        hom := Julia.Oscar._iso_oscar_gap(pol.parent.base_ring);
+        x := Indeterminate(hom.header.codomain,"x");
```

Comment:
This code relies on special properties of `hom` that are not necessarily always true. Instead of accessing the components directly, better to use the official APIs. E.g. here:
```suggestion
        x := Indeterminate(Julia.codomain(hom),"x");
```